src/Control.ts

```diff
@@ -0,0 +1,200 @@
+import { Connection } from './Connection';
+
+
+interface ControlProperties {
+    id?: string,
+    childControls?: Control[],
```

Comment:
I'm not sure if a common `childControls` going to universally work. There are controls that could have several child collections, like Grid having `columns` and `items` and there are controls without any children like Text.